src/components/navbar/Navbar.jsx

```diff
@@ -1,28 +1,39 @@
 import React from "react";
 import { Link, useNavigate } from "react-router-dom";
 import { useAuth } from "../../contexts/auth-context";
+import { useLike } from "../../contexts/like-context";
+import { useWatchLater } from "../../contexts/watch-later-context";
 import "./navbar.css";
 
 export default function Navbar() {
-
-  const {user, setUser} = useAuth()
- const navigate = useNavigate()
+  const { user, setUser } = useAuth();
+  const navigate = useNavigate();
+  const { like, setLike } = useLike();
+  const { watchLater, setWatchLater } = useWatchLater();
 
   const logoutHandler = () => {
-    setUser({token: null})
-    navigate("/")
-  }
+    setUser({ token: null });
+    setLike({ like: [] });
+    setWatchLater({ watchLater: [] });
```

Comment:
You can make use of reducer instead of useState. As it will be easy to handle different states together.